gems/sorbet-runtime/lib/types/enum.rb

```diff
@@ -372,7 +375,11 @@ def self.enums(&blk)
 
     orphaned_instances = T.must(@values) - @mapping.values
     if !orphaned_instances.empty?
-      raise "Enum values must be assigned to constants: #{orphaned_instances.map {|v| v.instance_variable_get('@serialized_val')}}"
+      serialized_vals = orphaned_instances.map do |instance|
+        serialized_val = instance.instance_variable_get('@serialized_val')
+        serialized_val.equal?(UNSET) ? 'unset' : serialized_val.inspect
+      end.join(', ')
```

Comment:
This is a bit awkward, but otherwise unset values will appear in the error message like `[#<Object:0x0000000101376e08>]`. That's because the `@serialized_val` instance variable hasn't yet been reassigned to a string representation of the constant name (which doesn't exist).

Reply:
You can fix this by declaring `UNSET` as

```ruby
UNSET = Module.new.freeze
```

instead of `Object.new.freeze`. This way, the `to_s` method of the `UNSET` object will be implemented by way of forwarding onto `UNSET.name`, which because the module has been assigned to a constant, will use that constant's name.

Reply:
That's much nicer, thanks for the suggestion.